dnf/base.py

```diff
@@ -2105,21 +2112,24 @@ def upgrade(self, pkg_spec, reponame=None):
         q = solution["query"]
         if q:
             wildcard = dnf.util.is_glob_pattern(pkg_spec)
+            installed = self.sack.query().installed()
+            obsoleters = q.filter(obsoletes=installed) \
```

Comment:
I would prefer to keep the `installed` (and also `obsoleters`) construction in the 
```python
if not wildcard and solution['nevra'] and solution['nevra'].name:
```
condition since they are used just there.

Reply:
Good point. It means that I deleted other places where I planned to use it.